### PR TITLE
[build] Fixed possibility to compile examples with C++03 standard

### PR DESCRIPTION
**Pull request by ethouris**
Fixed:

The examples were enforced to be compiled with C++11 standard even though they are written in C++03 or even in C style.

After the fix it is possible to compile the library and examples using a compiler that is not C++11 capable using options:

```
./configure --disable-apps --use-c++-std=03 --enable-examples
```

### CODE REVIEW

File: docs/DevelopersGuide.md [hunk 1]
```diff
@@ -77,12 +77,36 @@ Please see the following document for `configure` usage: [BuildOptions.md](./Bui
...
 to the target medium (UDP/SRT/`stdout`). See [srt-live-transmit.md](./srt-live-transmit.md) for more info.
-* `srt-file-transmit` — A sample application to transmit files with SRT.
-* `srt-tunnel` — A sample application to set up an SRT tunnel for TCP traffic. See [srt-tunnel.md](./srt-tunnel.md) for more info.
+* `srt-file-transmit` - A sample application to transmit files with SRT.
+* `srt-tunnel` - A sample application to set up an SRT tunnel for TCP traffic. See [srt-tunnel.md](./srt-tunnel.md) for more info.
 * `tests-srt` - unit testing application.
 
+## Language standard requirements
+
+The following convention for the language standard is used in this project:
```
Comment (stevomatthews) [inline docs/DevelopersGuide.md:88]:
```suggestion
The following conventions for the language standard are used in this project:
```

File: docs/DevelopersGuide.md [hunk 2]
```diff
@@ -77,12 +77,36 @@ Please see the following document for `configure` usage: [BuildOptions.md](./Bui
...
+
+1. The SRT library requires C++03 (also known as C++98) standard.
+2. The examples (to be enabled by `--enable-examples`) require either C++03 or C89 standard.
+3. The following remaining parts require C++11 standard:
```
Comment (stevomatthews) [inline docs/DevelopersGuide.md:92]:
```suggestion
3. The following  require C++11 standard:
```

File: docs/DevelopersGuide.md [hunk 3]
```diff
@@ -77,12 +77,36 @@ Please see the following document for `configure` usage: [BuildOptions.md](./Bui
...
+that does not support C++11 and you want to compile as many parts as possible,
+the simplest way is to use the following options:
+
+```
+./configure --disable-apps --use-c++-std=03 --enable-examples
+```
+
+Note also that several other options may require C++11 standard also for the library.
+For example, if you decide to `--enable-stdc++-sync`, it requires that the SRT library
+be compiled using C++11 standard.
```
Comment (stevomatthews) [inline docs/DevelopersGuide.md:108]:
```suggestion
Note also that there are several other options that, when enabled, may require that the SRT library be compiled using C++11 standard (`--enable-stdc++-sync` for example).
```

**Review by stevomatthews — COMMENTED** [review]
Minor edits.

File: CMakeLists.txt [hunk 4]
```diff
@@ -415,9 +415,9 @@ if (NOT ENABLE_CXX11)
 	if (ENABLE_STDCXX_SYNC)
 		message(FATAL_ERROR "ENABLE_STDCXX_SYNC is set, but C++11 is disabled by ENABLE_CXX11")
 	endif()
-	if (DEFINED USE_CXX_STD)
-		message(FATAL_ERROR "USE_CXX_STD can be set only when ENABLE_CXX11 is on")
-	endif()
+   #if (DEFINED USE_CXX_STD)
+   #	message(FATAL_ERROR "USE_CXX_STD can be set only when ENABLE_CXX11 is on")
+   #endif()
```
Comment (maxsharabayko) [inline CMakeLists.txt:420]: Can be removed

File: docs/DevelopersGuide.md [hunk 5]
```diff
@@ -77,12 +77,34 @@ Please see the following document for `configure` usage: [BuildOptions.md](./Bui
...
+   * unit tests (to be enabled by `--enable-unittests`)
+
+Note that C++11 standard will be enforced if you have enabled applications
+and haven't specified the C++ standard explicitly. When you have an old compiler
+that does not support C++11 and you want to compile as many parts as possible,
+the simplest way is to use the following options:
+
+```
+./configure --disable-apps --use-c++-std=03 --enable-examples
+```
```
Comment (maxsharabayko) [inline docs/DevelopersGuide.md:104]: CMake should be the default build system. TCL is not available on Windows by default.
```
cmake . -DENABLE_APPS=ON and so on
```

File: docs/DevelopersGuide.md [hunk 6]
```diff
@@ -77,12 +77,34 @@ Please see the following document for `configure` usage: [BuildOptions.md](./Bui
...
+
+Note also that there are several other options that, when enabled, may require that the SRT library be compiled using C++11 standard (`--enable-stdc++-sync` for example).
```
Comment (maxsharabayko) [inline docs/DevelopersGuide.md:106]: Convert to cmake

**Pull request merged**